compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -226,14 +227,48 @@ adjustDynFlags options@Options{..} dflags
   $ apply xopt_set xExtensionsSet
   $ apply xopt_unset xExtensionsUnset
   $ apply gopt_set (xFlagsSet options)
+  $ addPlatformFlags
+  $ addCppFlags
   dflags{
     mainModIs = mkModule primUnitId (mkModuleName "NotAnExistingName"), -- avoid DEL-6770
     debugLevel = 1,
-    ghcLink = NoLink, hscTarget = HscNothing -- avoid generating .o or .hi files
+    ghcLink = NoLink, hscTarget = HscNothing, -- avoid generating .o or .hi files
     {-, dumpFlags = Opt_D_ppr_debug `EnumSet.insert` dumpFlags dflags -- turn on debug output from GHC-}
+    ghcVersionFile = optGhcVersionFile
   }
-  where apply f xs d = foldl' f d xs
-
+  where
+    apply f xs d = foldl' f d xs
+    alterSettings f d = d { settings = f (settings d) }
+    addCppFlags = case optCppPath of
+        Nothing -> id
+        Just cppPath -> alterSettings $ \s -> s
+            { sPgm_P = (cppPath, [])
+            , sOpt_P = ["-P"]
+                -- We add "-P" here to suppress #line pragmas from the
+                -- preprocessor (hpp, specifically) because the daml
+                -- parser can't handle them. This is a non-issue right now
+                -- because ghcversion.h is empty, but if it weren't empty
+                -- it would result in #line pragmas. By suppressing these
+                -- pragmas, line numbers may be wrong up when using CPP.
+                -- Ideally we fix the issue with the daml parser and
+                -- then remove this flag.
+            }
+
+    -- We need to add platform info in order to run CPP. To prevent
+    -- .hi file incompatibilities, we set the platform the same way
+    -- for everyone even if they don't use CPP.
+    addPlatformFlags = alterSettings $ \s -> s
+        { sTargetPlatform = P.Platform
```

Comment:
What is the relationship between this and https://github.com/digital-asset/ghcide/blob/44b11667d8593ae84c48ca88e96f4f227deec096/src/Development/IDE/GHC/Util.hs#L95?

Reply:
Ah, I didn't see that. Maybe I should try to copy those platform values and see if that is defined enough for CPP. (I don't think it will be, because CPP exposes platform info as `-D` flags.)

Reply:
The `fakeDynFlags` platform settings does not work for `CPP`! It's missing required fields.

I changed `platformUnregisterised` to `True` to match `fakeDynflags` though. Thanks :-)